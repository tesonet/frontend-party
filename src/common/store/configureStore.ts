import { applyMiddleware, compose, createStore } from 'redux';
import middleware from './middleware';
import reducer from './reducer';
import { IAppState } from './types';

const devTool =
  process.env.TARGET === 'browser' && window.__REDUX_DEVTOOLS_EXTENSION__
    ? window.__REDUX_DEVTOOLS_EXTENSION__()
    : <T>(f: T): T => f;

const configureStore = (preloadedState?: IAppState) => {
  const enhancedCreateStore = compose<typeof createStore>(
    applyMiddleware(...middleware),
    devTool
  )(createStore);

  return enhancedCreateStore(reducer, preloadedState as IAppState);
};

export default configureStore;
