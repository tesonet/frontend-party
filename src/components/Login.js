import React, { Component } from "react";
import styled from "styled-components";
import { connect } from "react-redux";
import { auth } from "../store/actions/actions";
import { Redirect } from "react-router-dom";
import image from "../assets/images/riding-the-waves-3440x1440.jpg";
import { FaUser, FaLock } from "react-icons/fa";
import logo from "../assets/images/logo-testio.png";
import Form from "./styles/FormStyles";

const Wrapper = styled.div`
  align-items: center;
  justify-content: center;
  display: flex;
  height: 100vh;
  flex-direction: column;
  background: linear-gradient(rgba(25, 35, 62, 0.8), rgba(25, 35, 62, 0.8)),
    url(${image});
  background-size: cover;
  background-repeat: no-repeat;

  .wrapper-logo {
    margin-bottom: 50px;
  }
`;

class Login extends Component {
  state = {
    username: "",
    password: "",
    toServerList: false
  };

  changeHandler = e => {
    this.setState({ [e.target.name]: e.target.value });
  };

  submitHandler = async e => {
    e.preventDefault();

    await this.props.onAuth(this.state.username, this.state.password);
    if (this.props.token) {
      this.setState({ toServerList: true });
    }
  };

  render() {
    let errorMessage = null;
    if (this.props.error) {
      errorMessage = <p className="form-error-message">{this.props.error}</p>;
    }

    if (this.state.toServerList) {
      return <Redirect to="/server-list" />;
    }

    return (
      <Wrapper>
        <img className="wrapper-logo" src={logo} alt="Testio Logo" />
        <Form onSubmit={this.submitHandler}>
          <fieldset disabled={this.props.loading}>
            {errorMessage}
            <div className="input-container">
              <FaUser
                style={{
                  color: "#cccccc",
                  background: "white",
                  position: "absolute",
                  left: "13px",
                  top: "20px",
                  minWidth: "30px",
                  textAlign: "center"
                }}
              />
              <input
                required
                type="text"
                name="username"
                placeholder="Username"
                value={this.username}
                onChange={this.changeHandler}
              />
            </div>

            <div className="input-container">
              <FaLock
                style={{
                  color: "#cccccc",
                  background: "white",
                  position: "absolute",
                  left: "13px",
                  top: "20px",
                  minWidth: "30px",
                  textAlign: "center"
                }}
              />
              <input
                required
                type="password"
                name="password"
                placeholder="Password"
                value={this.password}
                onChange={this.changeHandler}
              />
            </div>

            <button className="form-button">Log In</button>
          </fieldset>
        </Form>
      </Wrapper>
    );
  }
}

const mapStateToProps = state => {
  return {
    isAuthenticated: state.token !== null,
    loading: state.loading,
    error: state.error,
    token: state.token
  };
};

const mapDispatchToProps = dispatch => {
  return {
    onAuth: (username, password) => dispatch(auth(username, password))
  };
};

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(Login);
