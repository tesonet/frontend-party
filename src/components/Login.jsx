import React from 'react';
import { withAlert } from 'react-alert';

import Icon from './Icon';
import InputGroup from './InputGroup';
import TesonetAPI from '../utilities/TesonetAPI';
import bindFunctions from '../utilities/bindFunctions';

import bigLogo from '../images/bigLogo.png';

class Login extends React.Component {
  constructor(props) {
    super(props);

    this.state = {
      password: '',
      username: '',
      loggingIn: false
    };

    bindFunctions(
      this,
      'onUsernameChange',
      'onPasswordChange',
      'onSubmit'
    );
  }

  onUsernameChange(event) {
    this.setState({username: event.target.value});
  }

  onPasswordChange(event) {
    this.setState({password: event.target.value});
  }

  onSubmit(event) {
    const {onLogin, alert} = this.props;

    event.preventDefault();

    this.setState({loggingIn: true});

    TesonetAPI
      .createToken(this.state.username, this.state.password)
      .then(
        onLogin,
        error => {
          this.setState({loggingIn: false});
          if (error instanceof TesonetAPI.HTTPError) {
            alert.error('Could not log in with provided username and password.');
          } else {
            alert.error('A network error has occured.');
          }
        }
      );
  }

  render() {
    const {username, password, loggingIn} = this.state;

    const submitDisabled = username === '' || password === '' || loggingIn;

    return (
      <div className="login-wrapper">
        <div className="container">
          <div className="login-container">
            <div className="login-inner">
              <img src={bigLogo} alt="testio." />
              <form onSubmit={this.onSubmit}>
                <InputGroup>
                  <Icon icon="fa-user" title="Username" />
                  <input
                    value={username}
                    onChange={this.onUsernameChange}
                    type="text"
                    placeholder="Username"
                    aria-label="Username"
                    className="form-control"
                  />
                </InputGroup>
                <InputGroup>
                  <Icon icon="fa-lock" title="Password" />
                  <input
                    value={password}
                    onChange={this.onPasswordChange}
                    type="password"
                    placeholder="Password"
                    aria-label="Password"
                    className="form-control"
                  />
                </InputGroup>
                <button type="submit" disabled={submitDisabled}>
                  {loggingIn ? 'Logging Inâ€¦' : 'Log In'}
                </button>
              </form>
            </div>
          </div>
        </div>
      </div>
    );
  }
}

export default withAlert(Login);
