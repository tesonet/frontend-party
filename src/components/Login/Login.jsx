import React, { memo } from 'react';
import { useDispatch } from 'react-redux';
import { useHistory } from 'react-router-dom';
import LoginForm from './LoginForm';
import actions from '../../actions/authActions';
import testioLogo from '../../assets/logo-testio.svg';

const Login = () => {
    const history = useHistory();
    const dispatch = useDispatch();
    const onSubmit = (values) => dispatch(actions.login(values, history));

    return (
        <div className="login-page">
            <img src={testioLogo} alt="" className="login-logo" />
            <LoginForm onSubmit={onSubmit} />
        </div>
    );
};

export default memo(Login);
