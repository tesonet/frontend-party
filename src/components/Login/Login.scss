.login {
  height         : 100%;
  width          : 100%;
  display        : flex;
  align-items    : center;
  background     : url('/static/images/bg.jpg') no-repeat center center fixed;
  background-size: cover;
  padding        : 0 2em;

  &__container {
    width: 100%;

    &-centered {
      display: block;
      margin : 0 auto;
    }
  }

  &__logo {
    display: block;
    margin : 0 auto 4rem;
  }
}