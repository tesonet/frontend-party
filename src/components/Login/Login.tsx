import React from "react";

const Login: React.FC = () => <div>Login</div>;

export default Login;
