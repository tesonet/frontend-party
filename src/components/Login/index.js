export { default as LoginBackground } from './LoginBackground';
export { default as LoginLogo } from './LoginLogo';
export { default as LoginForm } from './LoginForm';
export { default as LoginInput } from './LoginInput';
export { default as LoginButton } from './LoginButton';
export { default as LoginError } from './LoginError';
export { default as LoginRequiredError } from './LoginRequiredError';
