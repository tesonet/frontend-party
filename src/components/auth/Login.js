import React, { Component } from 'react';
import { Redirect } from 'react-router-dom';
import { connect } from 'react-redux';

import LoginLayout from './LoginLayout';
import FormField from '../utils/form/FormField';
import FormButton from '../utils/form/FormButton';
import {
    updateField,
    generateData,
    isFormValid
} from '../utils/form/formActions';
import { login } from '../../actions/authActions';

class Login extends Component {
    state = {
        formError: false,
        formErrorMessage: '',
        formdata: {
            username: {
                element: 'input',
                value: '',
                config: {
                    name: 'username',
                    type: 'text',
                    placeholder: 'Username'
                },
                icon: true,
                validation: {
                    required: true
                },
                valid: false,
                touched: false,
                errorMessage: ''
            },
            password: {
                element: 'input',
                value: '',
                config: {
                    name: 'password',
                    type: 'password',
                    placeholder: 'Password'
                },
                icon: true,
                validation: {
                    required: true
                },
                valid: false,
                touched: false,
                errorMessage: ''
            }
        }
    };

    updateForm = element => {
        const newFormdata = updateField(element, this.state.formdata);
        this.setState({
            formError: false,
            formErrorMessage: '',
            formdata: newFormdata
        });
    };

    onSubmit = e => {
        e.preventDefault();

        let dataToSubmit = generateData(this.state.formdata);
        let formIsValid = isFormValid(this.state.formdata);

        if (formIsValid) {
            this.props.login(dataToSubmit).then(() => {
                if (this.props.auth.message) {
                    this.setState({
                        formError: true,
                        formErrorMessage: this.props.auth.message
                    });
                }
            });
        } else {
            this.setState({
                formError: true,
                formErrorMessage: 'Something went wrong'
            });
        }
    };

    renderError = () =>
        this.state.formError && (
            <div className="form__error">{this.state.formErrorMessage}</div>
        );

    render() {
        if (this.props.auth.token) return <Redirect to="/servers" />;

        return (
            <LoginLayout>
                <form onSubmit={e => this.onSubmit(e)} className="form">
                    <FormField
                        id={'username'}
                        fieldData={this.state.formdata.username}
                        change={element => this.updateForm(element)}
                    />

                    <FormField
                        id={'password'}
                        fieldData={this.state.formdata.password}
                        change={element => this.updateForm(element)}
                    />

                    <FormButton type="submit" value="Log in" />
                    {this.renderError()}
                </form>
            </LoginLayout>
        );
    }
}

const mapStateToProps = state => ({
    auth: state.auth
});

export default connect(
    mapStateToProps,
    { login }
)(Login);
