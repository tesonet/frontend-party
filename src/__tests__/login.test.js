import React from 'react';
import configureStore from 'redux-mock-store';
import { mount } from 'enzyme';
import toJSON from 'enzyme-to-json';
import { Provider } from 'react-redux';
import thunk from 'redux-thunk';
import Login from '../pages/login';

const mockStore = configureStore([thunk]);
const historyMock = { push: jest.fn() };

const initialMocks = mockStore({
	userAuthentication: {
		wrongCredentials: false,
		loginPending: false,
		isLoggedIn: false,
	},
});
const wrongCredentialsMocks = mockStore({
	userAuthentication: {
		wrongCredentials: true,
		loginPending: false,
		isLoggedIn: false,
	},
});
const loginPendingMocks = mockStore({
	userAuthentication: {
		wrongCredentials: false,
		loginPending: true,
		isLoggedIn: false,
	},
});
const isLoggedInMocks = mockStore({
	userAuthentication: {
		wrongCredentials: false,
		loginPending: false,
		isLoggedIn: true,
	},
});

describe('<Login />', () => {
	test('displays all text correctly', () => {
		const wrapper = mount(
			<Provider store={initialMocks}>
				<Login />
			</Provider>
		);
		expect(wrapper.text()).toContain('Log In');
		expect(
			wrapper.find('.form__input--username').at(0).props().placeholder
		).toEqual('Username');
		expect(
			wrapper.find('.form__input--password').at(0).props().placeholder
		).toEqual('Password');
	});

	test('renders and matches snapshot', () => {
		const wrapper = mount(
			<Provider store={wrongCredentialsMocks}>
				<Login />
			</Provider>
		);
		expect(toJSON(wrapper.find('form'))).toMatchSnapshot();
	});

	test('redirects to /servers when user logs in', () => {
		const wrapper = mount(
			<Provider store={isLoggedInMocks}>
				<Login history={historyMock} />
			</Provider>
		);
		expect(historyMock.push.mock.calls[0]).toEqual(['/servers']);
	});

	test('displays warn if user havent typed anything in input', () => {
		const wrapper = mount(
			<Provider store={initialMocks}>
				<Login />
			</Provider>
		);
		wrapper.find('.form').simulate('submit');
		expect(wrapper.text()).toContain('Please enter your username');
		expect(wrapper.text()).toContain('Please enter your password');
	});

	test('displays warn message when credentials are incorrect', () => {
		const wrapper = mount(
			<Provider store={wrongCredentialsMocks}>
				<Login />
			</Provider>
		);
		expect(wrapper.text()).toContain('Incorrect username or password');
	});

	test('hides Log In text on button when authentication is pending', () => {
		const wrapper = mount(
			<Provider store={loginPendingMocks}>
				<Login />
			</Provider>
		);
		expect(wrapper.text()).not.toContain('Log In');
	});
});
