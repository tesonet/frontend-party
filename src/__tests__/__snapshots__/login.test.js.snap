// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Login /> renders and matches snapshot 1`] = `
<form
  className="form"
  method="post"
  onSubmit={[Function]}
>
  <label
    className="form__group"
  >
    <input
      className="form__input form__input--username"
      id="username"
      name="username"
      onChange={[Function]}
      placeholder="Username"
      type="text"
      value=""
    />
  </label>
  <label
    className="form__group"
  >
    <input
      className="form__input form__input--password"
      id="password"
      name="password"
      onChange={[Function]}
      placeholder="Password"
      type="password"
      value=""
    />
  </label>
  <Button
    className="btn btn--green"
    data-test="login-button"
    text="Log In"
    type="submit"
  >
    <button
      className="btn btn--green"
      type="submit"
    >
      Log In
    </button>
  </Button>
</form>
`;
